lib/ortho_functions.py

```diff
@@ -1816,6 +1826,78 @@ def overlap_check(geometry_wkt, spatial_ref, demPath):
 
     return overlap
 
+def check_image_overlap_with_gpkg(geometry_wkt, spatial_ref, gpkg_path):
+    """
+    Parameters:
+        image_geometry (wkt)
+        image_spatial_ref
+        gpkg_path (str): Path to the GeoPackage file.
+
+    Returns:
+        str: The value of the "dempath" field from the final overlapping layer, or None if no overlap is found.
+    """
+    # image properties
+    imageSpatialReference = spatial_ref.srs
+    image_geometry = ogr.CreateGeometryFromWkt(geometry_wkt)
+
+    # Open the GeoPackage dataset
+    dataset = gdal.OpenEx(gpkg_path, gdal.OF_VECTOR)
+    num_layers = dataset.GetLayerCount()
+    overlapping_layers = []
+
+    # Iterate over each layer in the dataset
+    for i in range(num_layers):
+        layer = dataset.GetLayerByIndex(i)
+        layer_spatial_ref = layer.GetSpatialRef()
+
+        # Check if the image geometry is in the same spatial reference as the current layer
+        if not imageSpatialReference.IsSame(layer_spatial_ref):
+            coordinate_transformer = osr.CoordinateTransformation(imageSpatialReference, layer_spatial_ref)
+            image_geometry_transformed = image_geometry.Clone()
+            image_geometry_transformed.Transform(coordinate_transformer)
+        else:
+            image_geometry_transformed = image_geometry
+
+        # Find the overlapping feature in the layer
+        feature = layer.GetNextFeature()
+        while feature:
+            feature_geometry = feature.GetGeometryRef()
+            if image_geometry_transformed.Intersects(feature_geometry):
+                overlapping_layers.append(layer)
+                break  # Break out of the loop since we found an overlap with this layer
+            feature = layer.GetNextFeature()
+
+    if not overlapping_layers:
+        return None  # No overlap found
+
+    # If there are multiple overlapping layers, choose the one where the image centroid is located
+    if len(overlapping_layers) > 1:
+        for layer in overlapping_layers:
+            feature = layer.GetNextFeature()
+            while feature:
+                feature_geometry = feature.GetGeometryRef()
+                if feature_geometry.Contains(image_geometry_transformed.Centroid()):
+                    layer.ResetReading()
+                    return feature.GetField("dempath")  # Return the value of the "dempath" field in the layer
+                feature = layer.GetNextFeature()
+
+    layer = overlapping_layers[0]
+    layer.ResetReading()
+    dempath = layer.GetNextFeature().GetField("dempath")
+
+    # check path compatibility with Windows
+    if platform.system() == "Windows":
+        dempath=dempath.replace("/mnt", "V:")
+        dempath=dempath.replace("/", "\\")
+
+    # Close the dataset
+    dataset = None
+
+    # If the centroid is not contained in any of the overlapping layers, return the "dempath" value from the first layer
```

Comment:
This comment is not exactly true, since the function returns None elsewhere if there are no overlapping features.